### ISSUE TITLE
question about "Training your own models"

### ISSUE THREAD

**Issue by gxtbluerain** [RenYang-home/OpenDVC#28]
when i train my models, some problems happen:
Operation received an exception:Status: 5, message: could not create a view primitive descriptor, in file tensorflow/core/kernels/mkl_slice_op.cc:300
i donot know how to solve it

**Comment by gxtbluerain** [comment]
Caused by op 'gradients/flow_motion/flow_cnn_4/concat_grad/Slice_2', defined at:
  File "F:\240629_OpenDVC\OpenDVC-master\OpenDVC_train_PSNR.py", line 108, in <module>
    train_MV = tf.train.AdamOptimizer(learning_rate=learning_rate).minimize(train_loss_MV, global_step=step)
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\training\optimizer.py", line 400, in minimize
    grad_loss=grad_loss)
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\training\optimizer.py", line 519, in compute_gradients
    colocate_gradients_with_ops=colocate_gradients_with_ops)
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\ops\gradients_impl.py", line 630, in gradients
    gate_gradients, aggregation_method, stop_gradients)
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\ops\gradients_impl.py", line 814, in _GradientsHelper
    lambda: grad_fn(op, *out_grads))
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\ops\gradients_impl.py", line 408, in _MaybeCompile
    return grad_fn()  # Exit early
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\ops\gradients_impl.py", line 814, in <lambda>
    lambda: grad_fn(op, *out_grads))
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\ops\array_grad.py", line 222, in _ConcatGradV2
    op, grad, start_value_index=0, end_value_index=-1, dim_index=-1)
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\ops\array_grad.py", line 150, in _ConcatGradHelper
    out_grads.append(array_ops.slice(grad, begin, size))
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\ops\array_ops.py", line 578, in slice
    return gen_array_ops._slice(input_, begin, size, name=name)
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\ops\gen_array_ops.py", line 7465, in _slice
    "Slice", input=input, begin=begin, size=size, name=name)
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\framework\op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\util\deprecation.py", line 488, in new_func
    return func(*args, **kwargs)
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\framework\ops.py", line 3274, in create_op
    op_def=op_def)
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\framework\ops.py", line 1770, in __init__
    self._traceback = tf_stack.extract_stack()

...which was originally created as op 'flow_motion/flow_cnn_4/concat', defined at:
  File "F:\240629_OpenDVC\OpenDVC-master\OpenDVC_train_PSNR.py", line 51, in <module>
    flow_tensor, _, _, _, _, _ = motion.optical_flow(Y0_com, Y1_raw, batch_size, Height, Width)
  File "F:\240629_OpenDVC\OpenDVC-master\motion.py", line 54, in optical_flow
    loss_4, flow_4 = loss(flow_3, im1_4, im2_4, 4)
  File "F:\240629_OpenDVC\OpenDVC-master\motion.py", line 27, in loss
    res = convnet(im1_warped, im2, flow, layer)
  File "F:\240629_OpenDVC\OpenDVC-master\motion.py", line 7, in convnet
    input = tf.concat([im1_warp, im2, flow], axis=-1)
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\ops\array_ops.py", line 1124, in concat
    return gen_array_ops.concat_v2(values=values, axis=axis, name=name)
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\ops\gen_array_ops.py", line 1032, in concat_v2
    "ConcatV2", values=values, axis=axis, name=name)
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\framework\op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\util\deprecation.py", line 488, in new_func
    return func(*args, **kwargs)
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\framework\ops.py", line 3274, in create_op
    op_def=op_def)
  File "E:\Anaconda\anaconda\envs\tensorflow\lib\site-packages\tensorflow\python\framework\ops.py", line 1770, in __init__
    self._traceback = tf_stack.extract_stack()

AbortedError (see above for traceback): Operation received an exception:Status: 5, message: could not create a view primitive descriptor, in file tensorflow/core/kernels/mkl_slice_op.cc:300
	 [[node gradients/flow_motion/flow_cnn_4/concat_grad/Slice_2 (defined at F:\240629_OpenDVC\OpenDVC-master\OpenDVC_train_PSNR.py:108)  = _MklSlice[Index=DT_INT32, T=DT_FLOAT, _kernel="MklOp", _device="/job:localhost/replica:0/task:0/device:CPU:0"](gradients/flow_motion/flow_cnn_4/conv2d/Conv2D_grad/Conv2DBackpropInput, ConstantFolding/gradients/flow_motion/flow_cnn_1/concat_grad/ConcatOffset-folded-2, gradients/flow_motion/flow_cnn_4/concat_grad/Shape_2, gradients/flow_motion/flow_cnn_4/conv2d/Conv2D_grad/Conv2DBackpropInput:1, DMT/_476, DMT/_477)]]

**Comment by RenYang-home** [comment]
Can you run the inference code? If the reference doesn't work neither, I guess that it is the problem of your environment.

**Comment by gxtbluerain** [comment]
> Can you run the inference code? If the reference doesn't work neither, I guess that it is the problem of your environment.

Thank you for your reply. The inference code is working, but currently, there is an error when running the code for training the PSNR models.

**Comment by RenYang-home** [comment]
I have never met this problem, so I am not so sure what exactly the problem could be, but I would still recommend you checking the environment, e.g., whether your installed TensorFlow is able to train other simple CNNs with concatenation layer?

**Comment by gxtbluerain** [comment]
> 我从来没有遇到过这个问题，所以我不太确定问题到底是什么，但我仍然建议你检查环境，例如，你安装的 TensorFlow 是否能够训练带有连接层的其他简单 CNN？

Thank you for your reply, what was the version of the python interpreter you were using in your code environment at that time?

**Comment by RenYang-home** [comment]
I think any python version that TF 1.12 could run with would be fine.

**Comment by gxtbluerain** [comment]
> I think any python version that TF 1.12 could run would be fine.

Thanks for the reply, it was due to environmental reasons, the issue has now been resolved, thanks again！
